__init__.py

```diff
@@ -92,51 +96,195 @@ def start_again(self):
+        much time is left for waiting. If user's time is evening
+        speaks when the shop opens in the morning. In other cases
+        just speaks shops info.
+        If user in open hours, speak the shop info. If shop open
+        and closes in one hour speaks how many minutes left.
+        Args:
+            shop_info (list): found shops on user's request
+            open (boolean): True - if shop is open
+        Returns:
+         3, None (to ask for another shop info)
+        Examples:
+            work time 9am-10pm
+            user's time 8am
+            Prompt: 'Shop is closed now. Opens in 1 hour'
+        """
+        for shop in shop_info:
+            work_time = shop['hours']
+            shop_name = shop['name']
+            day_time, hour, min = self.request_handler.curent_time_extraction()
+            parse_time = work_time.split('-')
+            open_time = int(re.sub('[^\d+]', '', parse_time[0]))
+            close_time = int(re.sub('[^\d+]', '', parse_time[1]))
+            # time left
+            wait_h = open_time - hour - 1
+            wait_min = 60 - min
+            if open:
+                if day_time[1] == 'pm' and 0 >= (close_time - hour) <= 1:
+                    self.speak(f'{shop_name} closes in {wait_min} minutes')
+                self.speak_shops([shop])
             else:
-                self.speak_dialog('all_locations')
-                return self.speak_shops(shop_info)
+                if day_time[1] == 'am' and hour < open_time:
+                    self.speak(f'{shop_name} is closed now. Opens in {wait_h} hour and {wait_min} minutes')
+                elif hour >= close_time:
+                    self.speak(f'{shop_name} is closed now. Shop opens at {open_time}')
+                self.speak_shops([shop])
+        return 3, None
+
+    def shops_by_time_selection(self, shop_info):
+        """
+        If user chose to select shops by time or
+        use like default selection. Selects open
+        shops.
+        Args:
+           shop_info (list): found shops on user's
+                               request
+        Returns:
+           shop_info (list): open shops
+        """
+        LOG.info(f"Shop by time selection {shop_info}")
+        open_shops = self.open_shops_search(shop_info)
+        if len(open_shops) >= 1:
+            return self.time_calculation(open_shops, True)
+        else:
+            return self.time_calculation(shop_info, False)
 
     def find_shop(self, user_request, mall_link):
+        """
+        When the intent is matched, user_request
+        variable contains the name of the shop.
+        The matching function get_shop_data() is
+        used to find the shop name in cache or
+        on the mall page.
+        If user's request is not None this function
+        can return several shops, one shop or empty
+        list.
+        If no shop was found asks user to repeat.
+        returns 1, user_request to continue the
```

Comment:
what do the `1` and `3` returns mean? Are these just arbitrary values?

Reply:
Yes, just arbitrary values